sql/core/src/main/scala/org/apache/spark/sql/package.scala

```diff
@@ -111,9 +111,10 @@ package object sql {
     }
   }
 
-  private val sparkCodePattern = Pattern.compile("org\\.apache\\.spark\\.sql\\." +
-      "(?:functions|Column|ColumnName|SQLImplicits|Dataset|DataFrameStatFunctions)" +
-      "(?:|\\..*|\\$.*)")
+  private val sparkCodePattern = Pattern.compile("(org\\.apache\\.spark\\.sql\\." +
+      "(?:functions|Column|ColumnName|SQLImplicits|Dataset|DataFrameStatFunctions|DatasetHolder)" +
+      "(?:|\\..*|\\$.*))" +
+      "|(scala\\.collection\\..*)")
```

Comment:
I added Scala collections classes as Spark classes because they might be mixed with them inside of Spark code.